unison-cli/src/Unison/Codebase/TranscriptParser.hs

```diff
@@ -426,8 +428,12 @@ run verbosity dir stanzas codebase runtime sbRuntime nRuntime config ucmVersion
             let f = Cli.LoadSuccess <$> readUtf8 (Text.unpack name)
              in f <|> pure Cli.InvalidSourceNameError
 
-      writeSourceFile :: ScratchFileName -> Text -> IO ()
-      writeSourceFile fp contents = do
+      writeSourceFile :: Bool -> ScratchFileName -> Text -> IO ()
```

Comment:
Would you explain the `Bool` arg — I'm looking at the spot where it's `False` and another spot where it's `True`, but I don't get it.

Reply:
`writeSourceFile` is called in two ways in the transcript runner:

1. When there's a `unison` block in the transcript it calls `writeSourceFile` to update the virtual file-system with the changes
2.  When a UCM command edits the scratchfile (e.g. `edit`).

However, within the transcript runner we need 1 to output as a `unison` block and 2 to defer a `unison:added-by-ucm` stanza until after the current `ucm` block and emit it there.

I can probably just separate off an `update virutal file system` function that's shared by `writeSource` and some new `unisonFenceBlock` function if that'd be clearer :)

Reply:
Okay, I  split off a `updateVirtualFile` function which should make things clearer :)

Reply:
Ok perfect, thanks.